pkg/graveler/graveler.go

```diff
@@ -1487,17 +1498,32 @@ func (g *Graveler) DeleteBatch(ctx context.Context, repository *RepositoryRecord
 	if len(keys) > DeleteKeysMaxSize {
 		return fmt.Errorf("keys length (%d) passed the maximum allowed(%d): %w", len(keys), DeleteKeysMaxSize, ErrInvalidValue)
 	}
-	var m *multierror.Error
+
 	err = g.safeBranchWrite(ctx, g.log(ctx).WithField("operation", "delete_keys"),
 		repository, branchID, func(branch *Branch) error {
-			var cachedMetaRangeID MetaRangeID // used to cache the committed branch metarange ID
+			commit, err := g.RefManager.GetCommit(ctx, repository, branch.CommitID)
+			if err != nil {
+				return err
+			}
+			cachedMetaRangeID := commit.MetaRangeID
+
+			deleteWorkers := len(keys)
+			if deleteWorkers > maxWorkerNum {
+				deleteWorkers = maxWorkerNum
+			}
+			keysChan := make(chan Key, len(keys))
+			var wg multierror.Group
```

Comment:
Usually wg is a WaitGroup; is-a multierror.Group a WaitGroup?

If not, I'd probably prefer
```suggestion
			var mg multierror.Group
```

Reply:
It is a waitGroup and this is the convention we are using in all usage instances